src/main/java/domain/Car.java

```diff
@@ -0,0 +1,44 @@
+package domain;
+
+// 차량의 이름, 차량의 위치를 가져야 함.
+public class Car {
+    // 자동차의 이름(name)과 위치(position)를 나타내는 변수
+    private final NumberGenerateStrategy strategy;
+    private final CarName name;
+    private int position;
+
+    // 생성자(constructor)
+    public Car(final NumberGenerateStrategy strategy, final String name) {
+        this(strategy, name, 0);
+    }
+
+    public Car(final NumberGenerateStrategy strategy,
+               final String name,
+               final int position) {
+        this.strategy = strategy;
+        this.name = CarName.from(name);
+        this.position = position;
+    }
+
+    public void move() {
+        final int value = strategy.generate();
+        if (value >= 4) {
+            position++;
+        }
+    }
+
+    public boolean isPositionSame(final int position) {
+        return this.position == position;
+    }
```

Comment:
[Comment]

오! Car가 내 위치가 다른 position이랑 same한지 확인하는 로직이 있다니 합릭적인 것 같아요!!

매개변수가 원시 타입인 것에 대해서는 어떻게 생각하시나요?

메서드는 어딘가에서 호출되기 때문에 정의하는 것이 잖아요.
그래서 매개변수에게 type이 있다는 것은 어떤 것을 input할지 힌트를 주니까 정말 다행인 일이에요.

position이라고 변수명을 적어놓긴 하셨지만, 자유도가 높은 타입인것 같아요.
저 로직이 호출되는 곳에선 정말 다양한 방법으로 int값이 넘어올 수 있겠죠.

 이런 방법들 중 하나를 선택한다면 어떨까요?
1. Car other를 매개변수로 한다
2. Position을 Wrapping 한다
```suggestion
    public boolean isPositionSame(final Car other) {
        return this.position == other.position;
    }
```

Reply:
[Approve]

변수를 의식적으로 final로 선언하는 것은 매우 좋은 자세라고 생각합니다!! 굿굿굿

Reply:
앞선 리뷰와 연결이 되는 내용이네요! 사용하는 변수가 '얼마나' 복잡한 조건이 있는가로 판단하기 보다는 position을 랩핑하는 것이 좋은 것 같습니다.

Reply:
Car other로 매개변수를 변경하면 Cars 클래스의 

public List<Car> findCarsByPosition(final int position) {
        return cars.stream()
                .filter(car -> car.isPositionSame(position))
                .toList();
    }

요 매서드 또한 영향을 받게 되는데 제 힘으로 해결되지 않는 오류가 계속 발생합니다...
어쩌면 position을 랩핑하는게 가장 나을 것 같습니다..